auth/proofOfPossession_test.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Cloud Security Client Go contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/pem"
+	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"os"
+	"path"
+	"testing"
+)
+
+func TestProofOfPossession_ParseCertHeader_edgeCases(t *testing.T) {
```

Comment:
I guess your solution is also working with chained certificates as you're using several libraries to parse the cert, in java I needed to introduce some parsing tweaks to format the fwd chained certs into a valid format. Maybe you can just double check that in go too.

Reply:
good point i think this needs to be checked as soon as we have a real one from k8s